src/components/transfers/EditForm.tsx

```diff
@@ -0,0 +1,299 @@
+  reason: yup.string().trim().min(1).max(300).required(),
+  documentId: yup.string().uuid().notRequired().nullable(),
+  targetDate: yup.date().min(new Date(), 'Date is invalid.').notRequired().nullable(),
+  approvedById: yup.string().uuid().notRequired().nullable(),
+  sourceVaultId: yup.string().uuid().required(),
+  sourceCampaignId: yup.string().uuid(),
+  targetVaultId: yup.string().uuid().required(),
+  targetCampaignId: yup.string().uuid(),
+})
+
+type props = {
+  transfer: TransferResponse
+  campaigns: CampaignResponse[]
+  vaults: VaultResponse[]
+  people: PersonResponse[]
+  id: string
+}
+
+export default function EditForm({ transfer, campaigns, vaults, people, id }: props) {
+  const { t } = useTranslation('transfer')
+
+  const router = useRouter()
+
+  const queryClient = useQueryClient()
+
+  const [sourceCampaignId, setSourceCampaignId] = useState(transfer.sourceCampaignId)
+  const [targetCampaignId, setTargetCampaignId] = useState(transfer.targetCampaignId)
+  const [sourceVaultId, setSourceVaultId] = useState(transfer.sourceVaultId)
+  const [targetVaultId, setTargetVaultId] = useState(transfer.targetVaultId)
+
+  const sourceVaults = vaults.filter((v) => v.campaignId === sourceCampaignId)
+  const targetVaults = vaults.filter((v) => v.campaignId === targetCampaignId)
+
+  const initialValues: TransferInput = {
+    status: transfer.status,
+    currency: transfer.currency,
+    amount: transfer.amount,
+    reason: transfer.reason,
+    documentId: transfer.documentId || '',
+    targetDate: dateParser(transfer.targetDate) || '',
+    approvedById: transfer.approvedById || '',
+    sourceCampaignId: sourceCampaignId,
+    sourceVaultId: sourceVaultId,
+    targetCampaignId: targetCampaignId,
+    targetVaultId: targetVaultId,
+  }
+
+  const mutation = useMutation<
+    AxiosResponse<TransferResponse>,
+    AxiosError<ApiErrors>,
+    TransferInput
+  >({
+    mutationFn: useEditTransfer(id),
+    onError: () => AlertStore.show(t('alerts:error'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('alerts:create'), 'success')
+      queryClient.invalidateQueries(endpoints.transfer.viewTransfer(id).url)
+      router.push(routes.admin.transfer.index)
+    },
+  })
+
+  function handleSubmit(values: TransferInput) {
+    const data: TransferInput = {
+      status: values.status,
+      currency: values.currency,
+      amount: values.amount,
+      reason: values.reason,
+      documentId: values.documentId ? values.documentId : null,
+      targetDate: values.targetDate ? new Date(values.targetDate) : null,
+      approvedById: values.approvedById ? values.approvedById : null,
+      sourceCampaignId: sourceCampaignId,
+      sourceVaultId: sourceVaultId,
+      targetCampaignId: targetCampaignId,
+      targetVaultId: targetVaultId,
+    }
+    if (
+      !data.sourceCampaignId ||
+      !data.targetCampaignId ||
+      !data.sourceVaultId ||
+      !data.targetVaultId
+    ) {
+      return
+    }
```

Comment:
This needs to be done via the validation

The validation should not allow you to submit without these values (make them `.required()`